eland/ml/exporters/es_gb_models.py

```diff
@@ -0,0 +1,449 @@
+        self.n_estimators_ = self.estimators_.shape[0]
+
+        for i in range(self.n_estimators_):
+            estimator = DecisionTreeType()
+            estimator.tree_ = self._trees[i + 1].tree_
+            estimator.n_features_in_ = self.n_features_in_
+            estimator.max_depth = self._max_depth
+            estimator.max_features_ = self.max_features_
+            self.estimators_[i, 0] = estimator
+
+    def _extract_common_parameters(self) -> None:
+        self.n_features_in_ = len(self.feature_names_in_)
+        self.max_features_ = self.n_features_in_
+
+    @property
+    def _max_depth(self) -> int:
+        return max(map(lambda x: x.max_depth, self._trees))
+
+    @property
+    def _n_outputs(self) -> int:
+        return self._trees[0].n_outputs
+
+    @property
+    def _definition(self) -> Mapping[Union[str, int], Any]:
+        return self._trained_model_result["trained_model_configs"][0]["definition"]
+
+    def _get_feature_names_in_(
+        self, preprocessors, feature_names, field_names
+    ) -> Tuple[List[str], Set[str]]:
+        input_field_names = set()
+
+        def add_input_field_name(preprocessor_type: str, feature_name: str) -> None:
+            if feature_name in feature_names:
+                input_field_names.add(preprocessor[preprocessor_type]["field"])
+
+        for preprocessor in preprocessors:
+            if "target_mean_encoding" in preprocessor:
+                add_input_field_name(
+                    "target_mean_encoding",
+                    preprocessor["target_mean_encoding"]["feature_name"],
+                )
+            elif "frequency_encoding" in preprocessor:
+                add_input_field_name(
+                    "frequency_encoding",
+                    preprocessor["frequency_encoding"]["feature_name"],
+                )
+            elif "one_hot_encoding" in preprocessor:
+                for feature_name in preprocessor["one_hot_encoding"][
+                    "hot_map"
+                ].values():
+                    add_input_field_name("one_hot_encoding", feature_name)
+
+        for field_name in field_names:
+            if field_name in feature_names and field_name not in input_field_names:
+                input_field_names.add(field_name)
+
+        return feature_names, input_field_names
+
+    def fit(self, X, y, sample_weight=None, monitor=None) -> None:
+        """
+        Override of the sklearn fit() method. It does nothing since Elastic ML models are
+        trained in the Elastic Stack or imported.
+        """
+        # Do nothing, model if fitted using Elasticsearch API
+        pass
+
+    @property
+    @abstractmethod
+    def analysis_type(self):
+        """
+        Type of the data frame analysis. It can be classification or regression.
+        """
+        pass
+
+    @abstractmethod
+    def _initialize_init_(self) -> Union[DummyClassifier, DummyRegressor]:
```

Comment:
Is this needed on the ABC? Looks like it's only used in implementations.

Reply:
You are right. I removed both abstract method definitions.